// scrapfly-sdk/build/src/logger.js
var LogLevel;
(function (LogLevel) {
    LogLevel["DEBUG"] = "DEBUG";
    LogLevel["INFO"] = "INFO";
    LogLevel["WARN"] = "WARN";
    LogLevel["ERROR"] = "ERROR";
})(LogLevel || (LogLevel = {}));
class Logger {
    level;
    name;
    constructor(name, level = LogLevel.INFO) {
        this.name = name;
        this.level = level;
    }
    debug(...args) {
        if (this.level <= LogLevel.DEBUG) {
            console.log(...args);
        }
    }
    info(...args) {
        if (this.level <= LogLevel.INFO) {
            console.info(...args);
        }
    }
    warn(...args) {
        if (this.level <= LogLevel.WARN) {
            console.warn(...args);
        }
    }
    error(...args) {
        if (this.level <= LogLevel.ERROR) {
            console.error(...args);
        }
    }
    setLevel(level) {
        if (Object.values(LogLevel).includes(level)) {
            this.level = level;
        }
        else {
            console.error(`Invalid log level: ${level}`);
        }
    }
}
export default Logger;
export const log = new Logger('scrapfly');
